src/hp/derp/client_conn.rs

```diff
@@ -747,44 +688,130 @@ mod tests {
     }
 
     #[tokio::test]
-    async fn test_client_conn_reader_basic() -> Result<()> {
-        let (reader, mut writer) = tokio::io::duplex(1024);
-        let (send_pong_s, mut send_pong_r) = mpsc::channel(10);
-        let (server_channel_s, mut server_channel_r) = mpsc::channel(10);
+    async fn test_client_conn_io_basic() -> Result<()> {
+        tracing_subscriber::registry()
+            .with(tracing_subscriber::fmt::layer().with_writer(std::io::stderr))
+            .with(EnvFilter::from_default_env())
+            .try_init()
+            .ok();
```

Comment:
could i nudge you to try out https://github.com/n0-computer/iroh/blob/6df66f479aac274f8231aaabbe3d11f07521b6a4/src/test_utils.rs#L16 ?  It seems there might still be a bug with it so if it blows up for you stick with this.  But otherwise it brings some benefits.

Reply:
what's the latest status on this? Having trouble grasping it based on the latest discord convo on logging.

Reply:
since this test uses a single-threaded tokio runtime it Should Just Work (tm) with `crate::test_utils::setup_logging`.  it's the multi-threaded runtimes that are still struggling.